### PR TITLE
Watcher and Client

### PR DESCRIPTION
**Pull request by rodrigolc**

### CODE REVIEW

File: workload/src/x509context.h [hunk 1]
```diff
@@ -0,0 +1,25 @@
+#ifndef INCLUDE_WORKLOAD_X509CONTEXT_H
+#define INCLUDE_WORKLOAD_X509CONTEXT_H
+
+#include "../../bundle/x509bundle/src/set.h"
+#include "../../svid/x509svid/src/svid.h"
+
+typedef struct workloadapi_X509Context {
+    x509svid_SVID **SVIDs;
```
Comment (otaviolcs3) [inline workload/src/x509context.h:8]: Replace "SVIDs" for "svids" and "Bundles" for "bundles".

File: workload/src/watcher.h [hunk 2]
```diff
@@ -1,103 +1,104 @@
...
+
+    // thread spun to wait on updates
+    thrd_t watcherThread;
     int threadError;
 
-    workloadapi_X509Callback x509Callback; //function called with updated x509Context
-    // jwtBundleSetFunc_t* jwtBundleSetUpdateFunc ; //function called with updated x509Context
-    
+    // function called with updated x509Context
+    workloadapi_X509Callback x509Callback;
```
Comment (otaviolcs3) [inline workload/src/watcher.h:54]: Replace "x509Callback" for "x509callback"

File: workload/src/watcher.h [hunk 3]
```diff
@@ -1,103 +1,104 @@
...
     mtx_t closeMutex;
     bool closed;
-    err_t closeError; //needed?
-    
-    thrd_t watcherThread; //thread spun to wait on updates
+    err_t closeError;
+
+    // thread spun to wait on updates
+    thrd_t watcherThread;
     int threadError;
```
Comment (otaviolcs3) [inline workload/src/watcher.h:51]: Replace "closeError", "watcherThread" and "threadError" for "close_error", "watcher_thread" and "thread_error", respectively

File: workload/src/watcher.h [hunk 4]
```diff
@@ -1,103 +1,104 @@
...
     bool updated;
-    
-    ///TODO: needed?
-    err_t updateError;
 
+    err_t updateError;
 
-    //Close sync
+    // Close sync
     mtx_t closeMutex;
```
Comment (otaviolcs3) [inline workload/src/watcher.h:45]: Replace "ownsClient", "updateMutex", "updateCond", "updateError", "closeMutex" for "owns_client", "update_mutex", "update_error" and "close_mutex", respectively

File: workload/src/watcher.h [hunk 5]
```diff
@@ -1,103 +1,104 @@
...
-{
-    ///TODO: add actual client type
-    void* client;
-    ///TODO: add actual client option type
-    void** clientOptions;
+typedef struct workloadapi_Client workloadapi_Client;
+typedef void (*workloadapi_ClientOption)(workloadapi_Client *, void *);
+typedef struct workloadapi_WatcherConfig {
+    workloadapi_Client *client;
+    workloadapi_ClientOption *clientOptions;
```
Comment (otaviolcs3) [inline workload/src/watcher.h:24]: Replace "clientOptions" for "client_options"

File: workload/src/client.h [hunk 6]
```diff
@@ -1,27 +1,116 @@
...
+typedef struct workloadapi_Watcher workloadapi_Watcher;
+
+typedef struct workloadapi_Client {
+    workloadapi_Stub stub;
+    bool ownsStub;
+    string_arr_t headers;
+    string_t address;
+    bool closed;
+    mtx_t closedMutex;
+    cnd_t closedCond;
```
Comment (otaviolcs3) [inline workload/src/client.h:26]: Replace "ownsStub", "closedMutex", "closedCond" for "owns_stub", "closed_mutex" and "closed_cond", respectively

**Review by otaviolcs3 — CHANGES_REQUESTED** [review]
See comments on files.

**Review by otaviolcs3 — APPROVED** [review]

**Comment by rodrigolc** [comment]
Merged by @otaviolcs3

**Pull request closed without merging**